backbone/src/main/java/org/researchstack/backbone/ui/step/layout/StepLayout.java

```diff
@@ -1,13 +1,23 @@
 package org.researchstack.backbone.ui.step.layout;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.view.View;
 
 import org.researchstack.backbone.result.StepResult;
+import org.researchstack.backbone.result.TaskResult;
 import org.researchstack.backbone.step.Step;
 import org.researchstack.backbone.ui.callbacks.StepCallbacks;
 
 public interface StepLayout
 {
-    void initialize(Step step, StepResult result);
+    /**
+     * @param step Step to be related to this StepLayout
+     * @param result the StepResult for this step, if one already exists
+     * @param taskResult The TaskResult object, if this StepLayout belongs to a Task
+     */
+    void initialize(@NonNull Step step,
+                    @Nullable StepResult result,
+                    @Nullable TaskResult taskResult);
```

Comment:
Those are some good thoughts Josh.  I think I'm going to revert this change, and push some more logic into the OnboardingTaskActivity.  That activity seems like the one that should be passing around certain info to it's StepLayouts, and it is also the one that knows the TaskResult.